lisa/tools/sysbench.py

```diff
@@ -0,0 +1,481 @@
+        return True
+
+    def install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        posix_os.install_packages("sysbench")
+
+        return self._check_exists()
+
+    def __run_sysbench_test(
+        self,
+        test_type: str,
+        threads: int,
+        events: int,
+        time_limit: int,
+        validation: str,
+        extra_args: str = "",
+    ) -> ExecutableResult:
+        args = (
+            f"{test_type} run --threads={threads} --events={events}"
+            f" --validate={validation} --percentile=95"
+            f" --verbosity=5 --histogram=on"
+            f" --time={time_limit} --debug=on {extra_args}"
+        )
+
+        result: ExecutableResult = self.run(
+            args,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="Sysbench Test failed",
+        )
+        return result
+
+    def run_cpu_perf(
+        self,
+        test_result: Any,
+        threads: int = 1,
+        events: int = 0,
+        cpu_max_prime: int = 10000,
+        time_limit: int = 10,
+    ) -> None:
+        extra_args: str = f" --cpu-max-prime={cpu_max_prime}"
+        perf_msg: Dict[Any, Any] = {}
+        testcase_name: str = inspect.stack()[1][3]
+        res = self.__run_sysbench_test(
+            test_type="cpu",
+            threads=threads,
+            events=events,
+            time_limit=time_limit,
+            validation="on",
+            extra_args=extra_args,
+        )
+        perf_msg = self.__process_cpu_perf_result(res.stdout)
+        perf_msg["test_case_name"] = testcase_name
+
+        self.__send_subtest_msg(
+            message_type=CPUPerformanceMessage,
+            test_name=testcase_name,
+            test_result=test_result,
+            other_fields=perf_msg,
+        )
+
+    def run_fileio_perf(
+        self,
+        operation: str,
+        test_result: Any,
+        threads: int = 1,
+        events: int = 0,
+        time_limit: int = 10,
+        file_io_mode: str = "sync",
+        file_fsync_all: str = "off",
+        file_fsync_mode: str = "fsync",
+        file_fsync_end: bool = False,
+        total_file: int = 128,
+        block_size_in_kb: int = 16,
+        file_total_size_in_gb: int = 2,
+        file_async_backlog: int = 128,
+        file_fsync_freq: int = 100,
+        file_merged_requests: int = 0,
+        file_rw_ratio: float = 1.5,
+    ) -> None:
+        total_size = file_total_size_in_gb * 1024 * 1024 * 1024
+        block_size = block_size_in_kb * 1024
+        valid_io_mode: List[str] = ["sync", "async", "mmap"]
+        valid_fsync_mode: List[str] = ["fsync", "fdatasync"]
+        valid_test_mode: Dict[str, str] = {
+            "seqwr": "write",
+            "seqrd": "read",
+            "rndrd": "read",
+            "rndwr": "write",
+            "seqrewr": "all",
+            "rndrw": "all",
+        }
+        perf_msg: Dict[Any, Any] = {}
+        testcase_name: str = inspect.stack()[1][3]
```

Comment:
How about put io mode into test case name? I'm not sure if it's easier to compare in dashboard, if the values are separated.